python/feathub/dsl/ast.py

```diff
@@ -218,6 +351,24 @@ def __init__(
         self.results = results
         self.default = default
 
+    def eval_dtype(self, variable_types: Dict[str, DType]) -> DType:
+        result_types = set(
+            [result_expr.eval_dtype(variable_types) for result_expr in self.results]
+        )
+
+        if not isinstance(self.default, NullNode):
+            result_types.add(self.default.eval_dtype(variable_types))
+
+        if len(result_types) == 1:
+            return result_types.pop()
+
+        result_type = _get_higher_precision_type(*result_types)
+        if result_type is None:
```

Comment:
Do we need this check?

Reply:
I think we need this check for the same reason above.